zulipterminal/ui_tools/boxes.py

```diff
@@ -1144,6 +1105,43 @@ def soup2markup(cls, soup: Any, metadata: Dict[str, Any],
                     f"Original text was {raw_text.strip()}"
                 )
                 metadata['time_mentions'].append((time_string, source_text))
+            elif css_classes:
```

Comment:
Yes, I just realized that the check for `css_classes` is unnecessary now. Thought that having all classes in the same place would be better. Even the speed gains won't be that helpful as very few messages out of many being loaded will recursively use soup2mark.
Re saving line-length from not indenting would still lead to re-splitting. Still, linear if-else seems like the right way to go considering ordering constraints.

Reply:
I think the ordering should be done according to most used tags/classes so that the loop ends faster, which it is not doing right now.
| Currently | Proposed |
| :----------:  | :-----------: |
| NavString | NavString |
| Unrendered Div classes | Paragraph |
| Custom emoji | Bold-Italics |
| Unrendered Tags | mentions |
| Paragraph | block quotes |
| emoji | Links |
| math | unrendered-tags(has images) |
| mentions | emoji |
| links | custom emoji |
| blockquote | code |
| code | codeblock|
| block-code | lists list-items |
| bold-italics | table |
| lists and list-items | unrendered div(inline img, inline ref, twitter) |
| table | math |
| time | time |